@nrwl/esbuild watch mode does not output assets

<!-- Please do your best to fill out all of the sections below! -->

## Current Behavior
When using the @nrwl/esbuild:esbuild in normal mode, the assets are being added to the dist.
When using watch mode, the assets are not being added

## Expected Behavior
Assets should be getting added in watch mode as well

## Steps to Reproduce
I'll try to get one up asap and post back here

### Environment
Node : 16.17.0
   OS   : linux x64
   pnpm : 7.14.0

   nx : 15.0.4
   @nrwl/angular : 15.0.4
   @nrwl/cypress : 15.0.4
   @nrwl/detox : Not Found
   @nrwl/devkit : Not Found
   @nrwl/esbuild : 15.0.4
   @nrwl/eslint-plugin-nx : 15.0.4
   @nrwl/expo : Not Found
   @nrwl/express : Not Found
   @nrwl/jest : 15.0.4
   @nrwl/js : Not Found
   @nrwl/linter : 15.0.4
   @nrwl/nest : Not Found
   @nrwl/next : Not Found
   @nrwl/node : Not Found
   @nrwl/nx-cloud : Not Found
   @nrwl/nx-plugin : Not Found
   @nrwl/react : Not Found
   @nrwl/react-native : Not Found
   @nrwl/rollup : Not Found
   @nrwl/schematics : Not Found
   @nrwl/storybook : Not Found
   @nrwl/web : Not Found
   @nrwl/webpack : Not Found
   @nrwl/workspace : Not Found
   typescript : 4.8.4
   ---------------------------------------
   Local workspace plugins:
   ---------------------------------------
   Community plugins:

@DibyodyutiMondal I must be doing something wrong. I am not able to run custom executor. Getting this error 

```

✘ [ERROR] Cannot use "external" without "bundle"

XXXXXXXXXXXXXXXXXX/node_modules/@nrwl/esbuild/src/executors/esbuild/esbuild.impl.js:111
                const success = results.every((r) => { var _a; return ((_a = r.errors) === null || _a === void 0 ? void 0 : _a.length) === 0; }) && !hasTypeErrors;
                                                                               ^

TypeError: Cannot read properties of undefined (reading 'errors')
    at XXXXXXXXXXXXXXXXX/packages/esbuild/src/executors/esbuild/esbuild.impl.ts:166:34
    at Array.every (<anonymous>)
    at XXXXXXXXXXXXXXXXX//packages/esbuild/src/executors/esbuild/esbuild.impl.ts:166:19
    at Generator.next (<anonymous>)
    at fulfilled (XXXXXXXXXXXXXXXXX//node_modules/tslib/tslib.js:115:62)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
Node.js v18.13.0
```

--- a/e2e/esbuild/src/esbuild.test.ts
+++ b/e2e/esbuild/src/esbuild.test.ts
@@ -13,6 +13,7 @@ import {
   packageInstall,
   rmDist,
   runCommandUntil,
+  waitUntil,
 } from '@nrwl/e2e/utils';
 
 describe('EsBuild Plugin', () => {
@@ -67,6 +68,31 @@ describe('EsBuild Plugin', () => {
     expect(() => runCLI(`build ${myPkg}`)).toThrow();
     expect(() => runCLI(`build ${myPkg} --skipTypeCheck`)).not.toThrow();
     expect(runCommand(`node dist/libs/${myPkg}/index.js`)).toMatch(/Bye/);
+    // Reset file
+    updateFile(
+      `libs/${myPkg}/src/index.ts`,
+      `
+      console.log('Hello');
+    `
+    );
+
+    /* Test that watch mode copies assets on start, and again on update.
+     */
+    updateFile(`libs/${myPkg}/assets/a.md`, 'initial a');
+    const watchProcess = await runCommandUntil(
+      `build ${myPkg} --watch`,
+      (output) => {
+        return output.includes('watching for changes');
+      }
+    );
+    readFile(`dist/libs/${myPkg}/assets/a.md`).includes('initial a');
+    updateFile(`libs/${myPkg}/assets/a.md`, 'updated a');
+    await expect(
+      waitUntil(() =>
+        readFile(`dist/libs/${myPkg}/assets/a.md`).includes('updated a')
+      )
+    ).resolves.not.toThrow();
+    watchProcess.kill();
   }, 300_000);
 
   it('should support bundling everything or only workspace libs', async () => {

--- a/packages/js/src/utils/assets/index.ts
+++ b/packages/js/src/utils/assets/index.ts
@@ -31,18 +31,19 @@ export async function copyAssets(
     callback:
       typeof options?.watch === 'object' ? options.watch.onCopy : undefined,
   });
+  const result: CopyAssetsResult = {
+    success: true,
+  };
+
   if (options.watch) {
-    const dispose = await assetHandler.watchAndProcessOnAssetChange();
-    return {
-      success: true,
-      stop: dispose,
-    };
-  } else {
-    try {
-      await assetHandler.processAllAssetsOnce();
-    } catch {
-      return { success: false };
-    }
-    return { success: true };
+    result.stop = await assetHandler.watchAndProcessOnAssetChange();
   }
+
+  try {
+    await assetHandler.processAllAssetsOnce();
+  } catch {
+    result.success = false;
+  }
+
+  return result;
 }

